assets/js/ui.js

```diff
@@ -0,0 +1,212 @@
+import { Carousel } from "./carousel.js";
+export const UI = {
+  get productList() {
+    return document.querySelector(".products-list");
+  },
+
+  get productsSection() {
+    return document.querySelector("#products");
+  },
+
+  get btnAddProduct() {
+    return document.querySelector("#add-product");
+  },
+
+  get btnSubmitProduct() {
+    return document.querySelector("#submit-product");
+  },
+
+  get btnDeleteProduct() {
+    return document.querySelector("#delete-product");
+  },
+
+  get btnEditProduct() {
+    return document.querySelector(".submit-edit");
+  },
+
+  get btnCancelModal() {
+    return document.querySelector(".cancel");
+  },
+
+  get modal() {
+    return document.querySelector("#modal");
+  },
+
+  get modalForm() {
+    return document.querySelector(".modal__form");
+  },
+
+  get modalDeleteControls() {
+    return document.querySelector(".modal__delete");
+  },
+
+  get modalContainer() {
+    return document.querySelector(".modal-container");
+  },
+
+  get inputTitle() {
+    return document.querySelector("#title");
+  },
+
+  get inputPrice() {
+    return document.querySelector("#price");
+  },
+
+  get inputImageUrl() {
+    return document.querySelector("#image-url");
+  },
+
+  get inputText() {
+    return document.querySelector("#text");
+  },
+
+  get inputHiddenId() {
+    return document.querySelector("#hidden-id");
+  },
+
+  get inputIsHighlighted() {
+    return document.querySelector("#isHighlighted");
+  },
+
+  showModal(state, type) {
+    state
+      ? UI.modal.classList.add("modal--show")
+      : UI.modal.classList.remove("modal--show");
+
+    UI.modalForm.style.display = type === "form" ? "block" : "none";
+    UI.modalDeleteControls.style.display = type === "delete" ? "block" : "none";
+  },
+
+  getValues() {
+    return {
+      id: parseInt(UI.inputHiddenId.value),
+      title: UI.inputTitle.value,
+      price: UI.inputPrice.value,
+      imageUrl: UI.inputImageUrl.value,
+      text: UI.inputText.value,
+      isHighlighted: UI.inputIsHighlighted.checked,
+    };
+  },
+
+  fillFormWithDataToEdit(data) {
+    UI.inputTitle.value = data.title;
+    UI.inputPrice.value = data.price;
+    UI.inputImageUrl.value = data.imageUrl;
+    UI.inputText.value = data.text;
+    UI.inputHiddenId.value = data.id;
+    UI.inputIsHighlighted.checked = data.isHighlighted;
+
+    UI.setButtonState("edit");
+  },
+
+  populateCarousel(data) {
```

Comment:
Consider having this code in Carousel class, especially when you already have the relevant query selectors there as well. 😉